src/wasm/wasm-stack.cpp

```diff
@@ -76,13 +86,29 @@ void BinaryInstWriter::visitCallIndirect(CallIndirect* curr) {
 }
 
 void BinaryInstWriter::visitLocalGet(LocalGet* curr) {
-  o << int8_t(BinaryConsts::LocalGet)
-    << U32LEB(mappedLocals[std::make_pair(curr->index, 0)]);
+  for (Index i = 0; i < func->getLocalType(curr->index).size(); ++i) {
```

Comment:
might be nice to capture the number of locals to a variable, like in `visitDrop` etc. - seems a little clearer to me. also I'm not sure I trust the optimizer.